research-prototype/Programs/Source/bsort.py

```diff
@@ -0,0 +1,77 @@
+from Compiler import types, library, instructions
```

Comment:
why do we have `asort.py` and `bsort.py`? can we drop one? if not, we need to make it obvious what the difference is. (we also need to move this one into the new `research_prototype/ipa` directory, if it's needed.)